### DISCUSSION TITLE
addRigidAttachment argument

### DISCUSSION THREAD

**Discussion by corredD** [NVIDIA-Omniverse/PhysX#198] [Q&A]
Hi,

could someone clarify what exactly is the position supposed to go in [addrigidattachments](https://nvidia-omniverse.github.io/PhysX/physx/5.2.1/_build/physx/latest/class_ext_gpu_1_1_px_particle_attachment_buffer.html?highlight=addrigidattachment#_CPPv4N6ExtGpu26PxParticleAttachmentBuffer18addRigidAttachmentEP12PxRigidActorK5PxU32RK6PxVec3P23PxConeLimitedConstraint) and how it affects the attachment ? Should it be the particle position relative to the rigid body ? But in the body of the function I can see the passed PxVec3 is transformed to local space : 
attachment.mLocalPose0 = PxVec4(body2Actor.transformInv(localPose), 0.f);

Thanks in advance

Ludo

**Comment by sschirm** [comment]
Hi @corredD, 
It should be the local position with respect to the PxRigidActor, which is the argument to the function. 
PxVec4(body2Actor.transformInv(localPose), 0.f);
Then transforms that local actor position to the PxRigidBody frame, which might not be the same, if the center of mass frame is not the identity transform.
